File: Odev-7-Capstone/Final/BackEndFinalProject/src/Infrastructure/SignalR/SignalRClient.cs
using Application.Common.Interfaces;
using Microsoft.AspNetCore.SignalR.Client;

namespace Infrastructure.SignalR
{
    public class SignalRClient : ISignalRClient
    {
        private HubConnection hubConnection;

        public SignalRClient()
        {
            hubConnection = new HubConnectionBuilder()
                .WithUrl("https://localhost:7016/Hubs/SeleniumLogHub") 
                .Build();
        }

        public async Task Connect()
        {
            await hubConnection.StartAsync();
        }

        public async Task SendLogNotification(string logMessage)
        {
            await hubConnection.InvokeAsync("SendLogNotificationAsync", logMessage);
        }
    }
}

File: Odev-7-Capstone/Final/FrontEndFinalProject/frontcrawler/src/utils/signalRClient.ts
import * as signalR from "@microsoft/signalr";

const connection = new signalR.HubConnectionBuilder()
    .withUrl("https://localhost:7016/Hubs/SeleniumLogHub")
    .build();

export const startSignalRConnection = async () => {
    try {
        await connection.start();
        console.log("SignalR connection started.");
    } catch (err) {
        console.error("Error starting SignalR connection:", err);
    }
};

export const stopSignalRConnection = async () => {
    try {
        await connection.stop();
        console.log("SignalR connection stopped.");
    } catch (err) {
        console.error("Error stopping SignalR connection:", err);
    }
};

export const onLogNotification = (callback: (logMessage: string) => void) => {
    connection.on("SendLogNotification", callback);
};
